docs/raising-events.md

```diff
@@ -9,58 +13,76 @@ public interface IRobot
 robot.FellInLove += Raise.With(sender: robot, e: someEventArgs);
 ```
 
+### VB.NET syntax
+
+```vbnet
+' Raise the event!
+AddHandler robot.FellInLove, Raise.With(someEventArgs) ' the "sender" will be robot
+
+' Use the overload for empty event args. Sender will be robot, args will be EventArgs.Empty
+AddHandler robot.FellInLove, Raise.WithEmpty()
+
+' Specify sender and event args explicitly:
+AddHandler robot.FellInLove, Raise.With(sender, someEventArgs)
+```
+
+## Raising `EventHandler<TEventArgs>`
+
 Events of type `EventHandler<TEventArgs>` can be raised in exactly the same way.
 
-Note that for the .NET 4.0 version of the FakeItEasy library, `TEventArgs` needs to extend `EventArgs` to support the above syntax.
+Note that for the .NET 4.0 version of the FakeItEasy library, `TEventArgs` needs to extend `EventArgs` to support the above syntax. Otherwise, the event must be raised as if it were a "free-form" event as described below.
+
+## "Free-form" events using arbitrary delegates
 
 It is also possible to raise events defined using a **custom delegate** (a.k.a
 "free-form delegate"), like these:
 
 ```csharp
-public delegate void CustomEventHandler(object sender, CustomEventArgs e);
 public delegate void FreeformEventHandler(int count);
+public delegate void CustomEventHandler(object sender, CustomEventArgs e); // considered a custom
+                                                                           // delegate in .NET 4.0
 …
-event CustomEventHandler CustomEvent;
 event FreeformEventHandler FreeformEvent;
+event CustomEventHandler CustomEvent;
 ```
 
-To do that, you can use `Raise.FreeForm.With`, which automatically infers the correct delegate type:
+### From C#
+To raise a free-form event from C#, use `Raise.FreeForm.With`, which automatically infers the correct delegate type:
 
 ```csharp
-fake.CustomEvent += Raise.FreeForm.With(fake, sampleCustomEventArgs);
 fake.FreeformEvent += Raise.FreeForm.With(7);
+fake.CustomEvent += Raise.FreeForm.With(fake, sampleCustomEventArgs);
 ```
 
-Notes on `Raise.Freeform.With`:
-- it only works in C#, not in VB.NET, due to language limitations;
-- it uses late binding, so you need a reference to the `Microsoft.CSharp` assembly in order to use it.
+Due to language limitations, `Raise.Freeform.With` does not work in VB.NET, and it uses late binding, so you need a reference to the `Microsoft.CSharp` assembly in order to use it.
 
-## VB.Net
+### From VB.NET
+To raise a free-form event from VB.NET, you must use `Raise.FreeForm<TEventHandler>.With`:
 
+```csharp
+fake.FreeformEvent += Raise.FreeForm<FreeformEventHandler>.With(7);
+fake.CustomEvent += Raise.FreeForm<CustomEventHandler>.With(fake, sampleCustomEventArgs);
 ```
-AddHandler robot.FellInLove, Raise.With(EventArgs.Empty)
-```
+
+Specifying the type of the event handler gets around the language restrictions in VB.NET.
+This method may also be used from C# if you want to avoid a reference to `Microsoft.CSharp`.

```

Comment:
I don't think anyone really cares about referencing Microsoft.CSharp, but some people prefer to avoid late binding. I'd suggest changing this to "if you don't want to rely on late binding" or something like that.